torch/_inductor/remote_cache.py

```diff
@@ -162,29 +163,31 @@ def __init__(
     # See if the cache contains `key`. Returns `None` if the value is not
     # present in the cache.
     def get(self, key: str) -> Optional[_T]:
-        sample = self._create_sample()
-        try:
-            result = self._get(key, sample)
-            cache_stats.get(type(self).__name__, result)
-        except Exception:
-            cache_stats.exception(type(self).__name__)
-            raise
-        self._log_sample(sample)
-        return result
+        with _WaitCounter("pytorch.remote_cache.get"):
```

Comment:
From a convo on another diff: are you instead planning to add this Waitcounter functionality to dynamo_timed? If so, can you sync w/ me on that?

And also: logging here will conflate different use cases, which may or may not be what we want. If we want to distinguish inductor remote cache from autotune remote cache, then we'd want to log "lower down"

Reply:
Sam and I talked directly, recording conclusions here for others to see :).

We currently have two parallel collection systems - dynamo_timed (which writes to a bunch of stuff). and WaitCounter (which is purely for counters).

Here we are purely trying to measure "remote network activity in cache", whether it is in inductor, or other things doing remote access. This is being done in WaitCounters due to where people want to consume it.

Not blocking this PR, but there is interest in writing to waitcounters from dynamo_timed, and unless it breaks stuff (i.e. multiple process dynamo timed has issues with events), we should probably have a span + a waitcounter for most spots within the compiler.

I am going to cut a PR for the base infrastructure (hopefully today, if not after thanksgiving). Then we'll work on consolidating the logging within compile.